rust/xaynet-analytics/src/data_combination/data_combiner.rs

```diff
@@ -128,14 +161,23 @@ where
         metadata: DataPointMetadata,
         events: Vec<AnalyticsEvent>,
     ) -> Vec<AnalyticsEvent> {
-        let start_of_period = self.get_start_of_period(metadata, metadata.period.n);
+        let start_of_period = self.get_start_of_period(metadata, None);
         self.filter_events_before_end_of_period(metadata.end, events)
             .into_iter()
             .filter(|event| event.timestamp > start_of_period)
             .collect()
     }
 
-    fn get_start_of_period(&self, metadata: DataPointMetadata, n_periods: u32) -> DateTime<Utc> {
+    fn get_start_of_period(
+        &self,
+        metadata: DataPointMetadata,
+        n_periods_override: Option<u32>,
+    ) -> DateTime<Utc> {
+        let n_periods = if n_periods_override == None {
+            metadata.period.n
+        } else {
+            n_periods_override.unwrap()
+        };
```

Comment:
this can be rewritten a bit and thus avoid the `unwrap`

```suggestion
        let n_periods = if let Some(n_periods) = n_periods_override {
            n_periods
        } else {
            metadata.period.n
        };
```

Reply:
thanks! will do